ops.c

```diff
@@ -0,0 +1,285 @@
+	return 0;
+}
+
+int sqfs_hl_op_create(const char* unused_path, mode_t unused_mode,
+		struct fuse_file_info *unused_fi) {
+	return -EROFS;
+}
+
+int sqfs_hl_op_release(const char *path, struct fuse_file_info *fi) {
+        free((sqfs_inode*)(intptr_t)fi->fh);
+        fi->fh = 0;
+        return 0;
+}
+
+int sqfs_hl_op_read(const char *path, char *buf, size_t size,
+                off_t off, struct fuse_file_info *fi) {
+        sqfs *fs;
+        sqfs_hl_lookup(&fs, NULL, NULL);
+        sqfs_inode *inode = (sqfs_inode*)(intptr_t)fi->fh;
+
+        off_t osize = size;
+        if (sqfs_read_range(fs, inode, off, &osize, buf))
+                return -EIO;
+        return osize;
+}
+
+ int sqfs_hl_op_readlink(const char *path, char *buf, size_t size) {
+	sqfs *fs;
+	sqfs_inode inode;
+	if (sqfs_hl_lookup(&fs, &inode, path))
+		return -ENOENT;
+	
+	if (!S_ISLNK(inode.base.mode)) {
+		return -EINVAL;
+	} else if (sqfs_readlink(fs, &inode, buf, &size)) {
+		return -EIO;
+	}	
+	return 0;
+}
+
+sqfs_hl *sqfs_hl_open(const char *path, size_t offset) {
+        sqfs_hl *hl;
+
+        hl = malloc(sizeof(*hl));
+        if (!hl) {
+                perror("Can't allocate memory");
+        } else {
+                memset(hl, 0, sizeof(*hl));
+                if (sqfs_open_image(&hl->fs, path, offset) == SQFS_OK) {
+                        if (sqfs_inode_get(&hl->fs, &hl->root, sqfs_inode_root(&hl->fs)))
+                                fprintf(stderr, "Can't find the root of this filesystem!\n");
+                        else
+                                return hl;
+                        sqfs_destroy(&hl->fs);
+                }
+
+                free(hl);
+        }
+        return NULL;
+}
+
+int sqfs_hl_op_listxattr(const char *path, char *buf, size_t size) {
+	sqfs *fs;
+	sqfs_inode inode;
+	int ferr;
+	
+	if (sqfs_hl_lookup(&fs, &inode, path))
+		return -ENOENT;
+
+	ferr = sqfs_listxattr(fs, &inode, buf, &size);
+	if (ferr)
+		return -ferr;
+	return size;
+}
+
+int sqfs_hl_op_statfs(const char *path, struct statvfs *st) {
+        sqfs_hl *hl = fuse_get_context()->private_data;
+        return sqfs_statfs(&hl->fs, st);
+}
+
+
+int sqfs_hl_op_getxattr(const char *path, const char *name,
+                char *value, size_t size
+#ifdef FUSE_XATTR_POSITION
+                , uint32_t position
+#endif
+                ) {
+        sqfs *fs;
+        sqfs_inode inode;
+        size_t real = size;
+
+#ifdef FUSE_XATTR_POSITION
+        if (position != 0) 
+                return -EINVAL;
+#endif
+
+        if (sqfs_hl_lookup(&fs, &inode, path))
+                return -ENOENT;
+
+        if ((sqfs_xattr_lookup(fs, &inode, name, value, &real)))
+                return -EIO;
+        if (real == 0)
+                return -sqfs_enoattr();
+        if (size != 0 && size < real)
+                return -ERANGE;
+        return real;
+}
+/** returns fuse_operations struct populated with function pointers to above operation functions*/
```

Comment:
Insert space between functions.

Also, each comment line should be 80 characters max. Break the comment into two separate lines.
```suggestion
}

/** returns fuse_operations struct populated with function pointers to above operation functions*/
```